src/exo/pyparser.py

```diff
@@ -1272,10 +959,14 @@ def parse_expr(self, e):
                 op = "@"
             else:
                 assert False, "unrecognized op"
+
             if op not in front_ops:
                 self.err(e, f"unsupported binary operator: {op}")
 
-            return PAST.BinOp(op, lhs, rhs, self.getsrcinfo(e))
+            if self.is_fragment:
+                return PAST.BinOp(op, lhs, rhs, self.getsrcinfo(e))
+            else:
+                return UAST.BinOp(op, lhs, rhs, self.getsrcinfo(e))
 
```

Comment:
A better design for this would be to write something like:
```
if is_fragment:
    self.AST = PAST
else:
    self.AST = UAST
```
and then just using (e.g.) `self.AST.BinOp` everywhere

Reply:
Ahh sounds good!